File: src/main/java/org/oddjob/dido/UnsupportedDataOutException.java
package org.oddjob.dido;

/**
 * Thrown when a {@link DataOut} doesn't support the required type.
 * 
 * @author rob
 *
 */
public class UnsupportedDataOutException extends DataException {
	private static final long serialVersionUID = 2010072700L;
	
	
	public UnsupportedDataOutException(Class<? extends DataOut> provider, 
			Class<? extends DataOut> required) {
		super(provider.getName() + " cannot provide " + required.getName());
	}

}

File: src/main/scala/org/oddjob/dido/UnsupportedDataOutException.scala
package org.oddjob.dido

/**
 * Thrown when a [[DataOut]] doesn't support the required type.
 * 
 * @author rob
 *
 */
class UnsupportedDataOutException(provider: Class[_ <: DataOut], required: Class[_ <: DataOut])
	extends DataException(provider.getName() + " cannot provide " + required.getName()) {

	private val serialVersionUID = 2010072700L

}